### DISCUSSION TITLE
What is your favorite JS framework?

### DISCUSSION THREAD

**Discussion by CarmelKalfa** [community/community#53628] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

**_Hey everyone_ 👋**

As a web developer, I have worked with several JS frameworks, and I am always curious to know what others in the community are using and their preferences. So, I thought it would be interesting to start a discussion on what your favorite JS framework is and why.

Personally, I have worked with React, Vue, and Angular, and each one has its strengths and weaknesses. However, I would have to say that React is my favorite JS framework. I find it to be lightweight, flexible, and easy to use. React's component-based architecture makes it easier to manage the UI, and its virtual DOM makes it faster and more efficient than other frameworks.

What about you? Which JS framework do you prefer, and why? Share your thoughts and experiences in the comments below. Let's have a lively discussion and learn from each other's experiences.

**Comment by NJacksonDev** [comment]
I like working with both React + Vue (lot of similarities). Want to also try out some of the latest ones - Svelte, Qwik, Solid.

**Comment by YasinDehfuli** [comment]
I think [Vue.js](https://vuejs.org/) , is the best

**Reply by rmdhfz** [reply]
why ?

**Comment by slyrackgo** [comment]
> ### Select Topic Area
> General
> 
> ### Body
> **_Hey everyone_ wave**
> 
> As a web developer, I have worked with several JS frameworks, and I am always curious to know what others in the community are using and their preferences. So, I thought it would be interesting to start a discussion on what your favorite JS framework is and why.
> 
> Personally, I have worked with React, Vue, and Angular, and each one has its strengths and weaknesses. However, I would have to say that React is my favorite JS framework. I find it to be lightweight, flexible, and easy to use. React's component-based architecture makes it easier to manage the UI, and its virtual DOM makes it faster and more efficient than other frameworks.
> 
> What about you? Which JS framework do you prefer, and why? Share your thoughts and experiences in the comments below. Let's have a lively discussion and learn from each other's experiences.

I would say Node.js. I am a blockchain developer and node.js is a perfect framework for this for example: encrypting data, working with the backend and all node.js packages are perfect. Though I also started yesterday  building project on react.js and it is pretty good experience considering that don't like front-end that much.

**Comment by CarmelKalfa** [comment]
thanks

**Comment by SRayen** [comment]
React is the best !

**Comment by imhul** [comment]
Svelte

**Reply by rmdhfz** [reply]
why ?

**Comment by santndev** [comment]
My favourite JavaScript framework is jQuery. It gained popularity for its simplicity and ease of use, allowing developers to quickly manipulate HTML documents and handle events. jQuery provided a concise syntax and a wide range of built-in functionalities that made it a go-to choice for many developers.

One aspect that some developers appreciated about jQuery is its ability to quickly generate executable files. Due to its lightweight nature and minimal setup requirements, developers found that jQuery-based projects could be built into executable files more quickly compared to some other frameworks.

**Reply by rmdhfz** [reply]
Are jQuery is framework javascript ?

**Reply by santndev** [reply]
No, jQuery is not a JavaScript framework. It is a JavaScript library that provides a simplified way to interact with HTML documents, handle events, perform animations, and manipulate the DOM (Document Object Model). jQuery simplifies common tasks in JavaScript and enhances cross-browser compatibility. However, it is not considered a full-fledged framework like Angular, React, or Vue.js.

But why do we need a framework just to quickly create executable scripts?

**Comment by infinite41207** [comment]
I think React is best!

**Comment by jackbereson** [comment]
nest.js (graphql) + next.js (ssr) = Backend Architecture + SSR Architecture = Nice works

**Comment by Hogunz** [comment]
React

**Comment by qhx0807** [comment]
(vue || react) & typescript 💯

**Comment by naaoufal** [comment]
ofc React JS

**Comment by DSDmark** [comment]
I also like **React.ts**. but, I heard lot about [**Astro**](https://astro.build/) which is **rust** base web framework. So, I'm planing to work on that.

**Comment by Hogunz** [comment]
What about three.js?

**Comment by cohhi** [comment]
Vue 😄

**Comment by denzeysenpai** [comment]
Angular, because it's just so clean

**Comment by SarthkDeshmukh** [comment]
Honestly? I'm primarily a Java backend developer, so I don't have a "favorite" the way a frontend-focused dev would. But from the backend side, when I've had to touch frontend work I've used ReactJS — and what I appreciate about it is that it thinks in components the same way I think in modular backend services.
